e2e/specs/st_tooltips_2.spec.js

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+describe("displays tooltips on text elements properly", () => {
+  before(() => {
+    cy.loadApp("http://localhost:3000/");
+    cy.prepForElementSnapshots();
+  });
+
+  it("Display text properly on tooltips on markdown", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 0)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on markdown."
+    );
+  });
+
+  it("Display text properly on tooltips on text", () => {
+    cy.get(`.stTextLabelWrapper > * > .stTooltipIcon`)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on text."
+    );
+  });
+
+  it("Display text properly on tooltips on latex", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 1)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on latex."
+    );
+  });
+
+  it("Display text properly on tooltips on caption", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 2)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on caption."
+    );
+  });
+
+  it("Display text properly on tooltips on title", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 3)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on title."
+    );
+  });
+
+  it("Display text properly on tooltips on header", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 4)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on header."
+    );
+  });
+
+  it("Display text properly on tooltips on subheader", () => {
+    cy.getIndexed(`.stMarkdown .stTooltipIcon`, 5)
+      .invoke("show")
+      .click();
+    cy.get("[data-testid=stMarkdownContainer]").should(
+      "contain",
+      "This is example tooltip displayed on subheader."
+    );
+  });
+
+  it("Tooltips match image snapshots", () => {
+    cy.getIndexed(".stTooltipContent", 0).matchImageSnapshot("stTooltipMarkdown")
+    cy.getIndexed(".stTooltipContent", 1).matchImageSnapshot("stTooltipText")
+    cy.getIndexed(".stTooltipContent", 2).matchImageSnapshot("stTooltipLatex")
+    cy.getIndexed(".stTooltipContent", 3).matchImageSnapshot("stTooltipCaption")
+    cy.getIndexed(".stTooltipContent", 4).matchImageSnapshot("stTooltipTitle")
+    cy.getIndexed(".stTooltipContent", 5).matchImageSnapshot("stTooltipHeader")
+    cy.getIndexed(".stTooltipContent", 6).matchImageSnapshot("stTooltipSubheader")
```

Comment:
Appreciate you adding e2e & e2e snapshot tests 🎉 
Strong preference to replace `.matchImageSnapshot` with our `.matchThemedSnapshots` custom command (see [here](https://github.com/streamlit/streamlit/blob/develop/frontend/cypress/support/commands.js#L88-L117) for details) because it generates snapshots covering both light & dark themes.

Reply:
Admittedly I gave the spec file a run using `.matchImageSnapshot`([run here](https://github.com/streamlit/streamlit/actions/runs/4308055177)) and looks like it works - but I'm okay with leaving it alone if it causes the tooltips to disappear